public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -458,29 +458,18 @@ function serviceMapQuery(request: DataQueryRequest<TempoQuery>, datasourceUid: s
       }
 
       const { nodes, edges } = mapPromMetricsToServiceMap(responses, request.range);
-      nodes.fields[0].config = {
-        links: [
-          makePromLink(
-            'Request rate',
-            `sum by (client, server)(rate(${totalsMetric}{server="\${__data.fields.id}"}[$__rate_interval]))`,
-            datasourceUid,
-            false
-          ),
-          makePromLink(
-            'Request histogram',
-            `histogram_quantile(0.9, sum(rate(${histogramMetric}{server="\${__data.fields.id}"}[$__rate_interval])) by (le, client, server))`,
-            datasourceUid,
-            false
-          ),
-          makePromLink(
-            'Failed request rate',
-            `sum by (client, server)(rate(${failedMetric}{server="\${__data.fields.id}"}[$__rate_interval]))`,
-            datasourceUid,
-            false
-          ),
-          makeTempoLink('View traces', `\${__data.fields[0]}`, '', tempoDatasourceUid),
-        ],
-      };
+      nodes.fields[0].config = getFieldConfig(
+        datasourceUid,
+        tempoDatasourceUid,
+        '__data.fields.id',
+        '__data.fields[0]'
+      );
+      edges.fields[0].config = getFieldConfig(
```

Comment:
What do you think about also including the client in the links for the edge menu? Since we know both, we can show really specific metrics.

Reply:
I'm not super convinced this is a good approach. If I'm understanding you correctly, this would mean there are now 8 links in the context menu. This would create a bigger issue with rendering the menu as it's already cutoff if you render it close to the bottom of the screen. Not to mention we are planning on separating out the `View traces` menu item into it's own section.. meaning this change would then mean we have 4 sections! Wdyt?

Reply:
Sorry, to clarify I mean add the client as a label to the metrics queries so that the linked metrics are showing exactly the rate/errors/etc. for that client, server pair. E.g.: `sum by (client, server)(rate(${totalsMetric}{server="api",client="app"}[$__rate_interval]))`

Reply:
I see. Added :)